src/calculation/DetailedResultsBuilder.ts

```diff
@@ -52,12 +52,32 @@ export function createPopulationValues(
       // create patient level population and stratifier results based on episodes
       episodeResults.forEach(episodeResult => {
         episodeResult.populationResults.forEach(popResult => {
-          createOrSetResult(
-            popResult.populationType,
-            popResult.criteriaExpression,
-            popResult.result,
-            populationResults
+          const measureObsPops = populationResults.find(
+            pop => pop.populationType == PopulationType.OBSERV && popResult.criteriaExpression == pop.criteriaExpression
           );
+          if (!measureObsPops) {
+            // only add observations if it is not undefined
+            if (popResult.observations?.[0]) {
+              createOrSetResult(
+                popResult.populationType,
+                popResult.criteriaExpression,
+                popResult.result,
+                populationResults,
+                popResult.observations as string[]
+              );
+            } else {
+              createOrSetResult(
+                popResult.populationType,
+                popResult.criteriaExpression,
+                popResult.result,
+                populationResults
+              );
+            }
```

Comment:
My issue when I tried to do that was that when I passed it is as undefined, then the property would exist as "undefined" rather than be excluded completely (what we want). I am open to other solutions that you can think of that ensure this behavior!